packages/@remirror/extension-annotation/src/annotation-extension.ts

```diff
@@ -15,8 +14,20 @@ import {
   SetAnnotationsAction,
   UpdateAnnotationAction,
 } from './actions';
-import type { Annotation, AnnotationData, AnnotationOptions } from './types';
-import { toAnnotation, toDecoration } from './utils';
+import { AnnotationState } from './annotation-plugin';
+import type { Annotation, AnnotationData, AnnotationOptions, AnnotationWithoutText } from './types';
+
+/**
+ * Computes a background color based on how many overlapping annotations are in
+ * a segment. The more annotations, the darker the background. This gives the
+ * illusion that annotations are above each other.
+ */
+function defaultGetStyle<A extends Annotation>(annotations: Array<AnnotationWithoutText<A>>) {
+  // Consider up to 5 overlapping annotations
+  const backgroundShade = Math.min(annotations.length, 5) / 5;
```

Comment:
Your comments are great and give a really good understanding of what’s happening. 

Maybe store the number 5 as a constant `MAX_OVERLAPPING_ANNOTATIONS` or something better napped in the outer scope.